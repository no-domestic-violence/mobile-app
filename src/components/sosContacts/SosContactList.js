/* eslint no-underscore-dangle: ['error', { 'allow': ['_id'] }] */
import React, { useContext, useEffect, useState } from 'react';
import { StyleSheet, View } from 'react-native';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import { faPen } from '@fortawesome/free-solid-svg-icons';
import { Button } from 'react-native-elements';
import { useIsFocused, useNavigation } from '@react-navigation/native';
import { Context as AuthContext } from '_state/AuthContext';
import appApiClient from '_api/appApiClient';
import { StyledButton, StyledButtonText } from 'styles/shared/StyledButton';

export default function SosContactList() {
  const navigation = useNavigation();
  const { state } = useContext(AuthContext);

  const [dataSource, setDataSource] = useState([]);
  const isFocused = useIsFocused();

  const hasFirstContact = dataSource.length !== 0;
  const hasSecondContact = dataSource.length === 2;

  useEffect(() => {
    getContacts();
    navigation.setParams({ id: '' });
  }, [isFocused]);

  const getContacts = async () => {
    try {
      const response = await appApiClient.get(
        `/users/${state.username}/contacts`,
      );
      setDataSource([...response.data.contacts]);
    } catch (error) {
      console.error(error);
    }
  };

  return (
    <>
      <View style={styles.contactContainer}>
        <Button
          title={
            hasFirstContact
              ? dataSource[0].name
              : 'Please add your trusted contact'
          }
          titleStyle={
            hasFirstContact ? styles.contactText : styles.contactPlaceholder
          }
          type="solid"
          raised
          iconContainerStyle={styles.icon}
          buttonStyle={styles.buttonText}
          onPress={() =>
            navigation.navigate('SosContactForm', {
              id: hasFirstContact && dataSource[0]._id,
            })
          }
          icon={<FontAwesomeIcon icon={faPen} />}
        />

        <Button
          titleStyle={
            hasSecondContact ? styles.contactText : styles.contactPlaceholder
          }
          title={
            hasSecondContact
              ? dataSource[1].name
              : 'Please add your trusted contact'
          }
          type="solid"
          raised
          iconContainerStyle={styles.icon}
          buttonStyle={styles.buttonText}
          onPress={() =>
            navigation.navigate('SosContactForm', {
              id: hasSecondContact && dataSource[1]._id,
            })
          }
          icon={<FontAwesomeIcon icon={faPen} />}></Button>
      </View>
      {dataSource.length > 0 && (
        <StyledButton style={styles.messageButtonContainer}>
          <StyledButtonText style={styles.messageButtonText}>
            Ask for help to your contacts
          </StyledButtonText>
        </StyledButton>
      )}
    </>
  );
}

const styles = StyleSheet.create({
  homeView: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  contactContainer: {
    height: '25%',
    width: '75%',
    flexDirection: 'column',
    justifyContent: 'space-between',
  },
  contactText: {
    color: '#000000',
    marginLeft: 10,
    fontWeight: '600',
  },
  contactPlaceholder: {
    color: '#000000',
    opacity: 0.34,
    marginLeft: 10,
    fontWeight: '400',
    fontSize: 16,
  },
  messageButtonContainer: {
    marginTop: 100,
    height: '10%',
    borderRadius: 41,
    backgroundColor: '#D65137',
    justifyContent: 'center',
  },
  messageButtonText: {
    fontSize: 16,
  },
  icon: {
    position: 'absolute',
    left: 10,
    fontSize: 14,
    paddingRight: 15,
  },
  buttonText: {
    backgroundColor: '#FEF8E3',
    padding: 18,
    borderRadius: 41,
  },
});
