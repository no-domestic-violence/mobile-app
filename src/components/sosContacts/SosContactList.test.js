import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import SosContactList from './SosContactList';

jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

describe('<SosContactList />', () => {
  it('should match snapshot', () => {
    const result = render(<SosContactList />).toJSON();
    expect(result).toMatchSnapshot();
  });
});
