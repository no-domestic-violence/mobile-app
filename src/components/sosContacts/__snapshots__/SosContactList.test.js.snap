// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SosContactList /> should match snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "flexDirection": "column",
        "height": "25%",
        "justifyContent": "space-between",
        "width": "75%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 41,
          "overflow": "visible",
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="first-contact"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FEF8E3",
              "borderColor": "#2089dc",
              "borderRadius": 41,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 18,
            }
          }
        >
          <
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f304",
                  "M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z",
                ],
                "iconName": "pen",
                "prefix": "fas",
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#000000",
                "fontSize": 18,
                "fontWeight": "600",
                "marginLeft": 10,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            defaultname
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 41,
          "overflow": "visible",
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="second-contact"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FEF8E3",
              "borderColor": "#2089dc",
              "borderRadius": 41,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 18,
            }
          }
        >
          <
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f304",
                  "M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z",
                ],
                "iconName": "pen",
                "prefix": "fas",
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#000000",
                "fontSize": 16,
                "fontWeight": "400",
                "marginLeft": 10,
                "opacity": 0.34,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Please add your trusted contact
          </Text>
        </View>
      </View>
    </View>
  </View>,
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#D65137",
        "borderRadius": 41,
        "fontWeight": "600",
        "height": "10%",
        "justifyContent": "center",
        "marginTop": 100,
        "opacity": 1,
        "overflow": "hidden",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 14,
            "fontWeight": "600",
            "overflow": "hidden",
            "paddingBottom": 10,
            "paddingLeft": 50,
            "paddingRight": 50,
            "paddingTop": 10,
            "textAlign": "center",
          },
          Object {
            "fontSize": 16,
          },
        ]
      }
    >
      Ask for help to your contacts
    </Text>
  </View>,
]
`;
